devito/passes/iet/mpi.py

```diff
@@ -368,3 +367,42 @@ def mpiize(graph, **kwargs):
         make_halo_exchanges(graph, mpimode=mpimode, **kwargs)
 
     make_reductions(graph, mpimode=mpimode, **kwargs)
+
+
+# *** Utilities
+
+def _make_cond_mapper(iet):
+    cond_mapper = MapHaloSpots().visit(iet)
+    return {hs: {i for i in v if i.is_Conditional and
+                 not isinstance(i.condition, GuardFactorEq)}
+            for hs, v in cond_mapper.items()}
+
+
+def ensure_control_flow(hs0, hs1, cond_mapper):
+    """
+    # If there are Conditionals involved, both `hs0` and `hs1` must be
+    # within the same Conditional, otherwise we would break the control
+    """
+    cond0 = cond_mapper.get(hs0)
+    cond1 = cond_mapper.get(hs1)
+
+    return cond0 != cond1
+
+
+def motion_rules():
```

Comment:
Perhaps define a `motion_rules = _motion_rules()`?

Reply:
this should not be a function

Reply:
And I'd call this simply "rules"

```
def _rule0(...):
     """...."""
     <impl>

def _rule1(...):
     """...."""
     <impl>

rules = (_rule0, _rule1)
```

Reply:
done